src/js/components/Options.js

```diff
@@ -3,32 +3,65 @@ import { observable } from "mobx"
 import { observer } from "mobx-react"
 import { isChrome, getSettings, setStorage } from "utils/browser"
 import ApiClient from "api/Client"
+import remoteServices from "../remoteServices"
+import { map, sortedUniqBy, filter } from "lodash"
+import { getHostOverridesFromSettings } from "../utils/settings"
 
 @observer
 class Options extends Component {
   @observable subdomain = ""
   @observable apiKey = ""
+  @observable hostOverrides = {}
   @observable errorMessage = null
   @observable isSuccess = false
+  @observable servicesHostOverrideList = []
+  @observable showHostOverrideOptions = false
 
   componentDidMount() {
-    getSettings(false).then(({ subdomain, apiKey }) => {
-      this.subdomain = subdomain || ""
-      this.apiKey = apiKey || ""
+    this.servicesHostOverrideList = sortedUniqBy(
+      map(
+        filter(remoteServices, (remoteService) => {
+          return remoteService.allowHostOverride
+        }),
+        (remoteService) => {
+          return {
+            name: remoteService.name,
+            host: remoteService.host,
+          }
+        },
+      ),
+      "name",
+    )
```

Comment:
I did so at first, using a key/value map, where it did not work because unlike the normal map/filter methods by lodash the fp variants skip extra parameters unless you explicitly convert every single function beforehand. That cluttered it up a lot, but you are right: after switching to an array of objects, this is now a possibility and more readable. Will adjust it :)